lib/demuxer.ex

```diff
@@ -0,0 +1,175 @@
+
+  defmodule State do
+    @type t :: %__MODULE__{
+            actions_buffer: list,
+            parser_acc: binary,
+            phase: :all_outputs_linked | :awaiting_linking,
+            continued_packets: %{},
+            tracks: list
+          }
+
+    defstruct actions_buffer: [],
+              parser_acc: <<>>,
+              phase: :awaiting_linking,
+              continued_packets: %{},
+              tracks: []
+  end
+
+  @impl true
+  def handle_init(_ctx, _options) do
+    {[], %State{}}
+  end
+
+  @impl true
+  def handle_playing(_context, state) do
+    {[{:demand, :input}], state}
+  end
+
+  @impl true
+  def handle_pad_added(Pad.ref(:output, id), _context, state) do
+    stream_format = {Pad.ref(:output, id), %RemoteStream{type: :packetized, content_format: Opus}}
+    state = %State{state | phase: :all_outputs_linked, tracks: state.tracks ++ [id]}
+
+    {[stream_format: stream_format], state}
+  end
+
+  @impl true
+  def handle_process(:input, %Buffer{payload: bytes}, context, state) do
+    unparsed = state.parser_acc <> bytes
+
+    {parsed, unparsed, continued_packets} =
+      Membrane.Ogg.Parser.parse(unparsed, state.continued_packets)
+
+    state = %State{state | parser_acc: unparsed, continued_packets: continued_packets}
+
+    {actions, state} = process_packets(parsed, state)
+
+    process_actions(actions, context, state)
+  end
+
+  @impl true
+  def handle_demand(Pad.ref(:output, _id), _size, :buffers, context, state)
+      when state.phase == :all_outputs_linked do
+    process_actions(state.actions_buffer, context, %State{state | actions_buffer: []})
+  end
+
+  @impl true
+  def handle_demand(Pad.ref(:output, _id), _size, :buffers, _context, state) do
+    {:ok, state}
+  end
+
+  defp process_actions(actions, context, state) do
+    demands = get_demands_from_context(context, state)
+
+    {sent_actions, buffered_actions} = classify_actions(actions, demands, state)
+
+    state = %State{state | actions_buffer: buffered_actions}
+
+    actions = sent_actions ++ demand_if_not_blocked(state)
+
+    {actions, state}
+  end
+
+  defp get_demands_from_context(context, state) do
+    Enum.reduce(state.tracks, %{}, fn track, acc ->
+      Map.put(acc, track, context.pads[Pad.ref(:output, track)].demand)
+    end)
+  end
+
+  defp process_packets(packets_list, state) do
+    Enum.reduce(packets_list, {[], state}, fn packet, {actions, state} ->
+      {new_actions, state} = process_packet(packet, state)
+      {actions ++ new_actions, state}
+    end)
+  end
+
+  defp process_packet(packet, state) do
+    cond do
+      packet.bos? -> process_bos_packet(packet, state)
+      packet.eos? and packet.payload == <<>> -> {[], state}
+      true -> process_data_packet(packet, state)
+    end
+  end
+
+  defp process_bos_packet(
+         %{payload: <<"OpusHead", _rest::binary>>} = packet,
+         state
+       ) do
+    new_track_action = {:notify_parent, {:new_track, {packet.track_id, :opus}}}
```

Comment:
We need to document this notification, e.g. https://github.com/membraneframework/membrane_mp4_plugin/blob/master/lib/membrane_mp4/demuxer/isom.ex#L30